src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,49 +1,108 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import classNames from 'classnames/bind';
 import { Loader } from '../Loader';
+import { Props } from './TodoModal.types';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+import { Error } from '../Error/Error';
+
+export const TodoModal: React.FC<Props> = ({
+  selectedTodo,
+  handleCloseButton,
+}) => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [hasError, setHasError] = useState(false);
+
+  const loadUser = async () => {
+    try {
+      const user = await getUser(selectedTodo.userId);
+
+      setSelectedUser(user);
+    } catch {
+      setHasError(true);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    loadUser();
+  }, []);
+
+  const { title, id, completed } = selectedTodo;
 
-export const TodoModal: React.FC = () => {
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
 
-      {true ? (
+      {isLoading && (
         <Loader />
-      ) : (
+      )}
+
+      {selectedUser && (
         <div className="modal-card">
           <header className="modal-card-head">
             <div
               className="modal-card-title has-text-weight-medium"
               data-cy="modal-header"
             >
-              Todo #2
+              {`Todo #${id}`}
             </div>
 
-            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
             <button
+              aria-label="close"
               type="button"
               className="delete"
               data-cy="modal-close"
+              onClick={handleCloseButton}
             />
           </header>
 
           <div className="modal-card-body">
             <p className="block" data-cy="modal-title">
-              quis ut nam facilis et officia qui
+              {title}
             </p>
 
             <p className="block" data-cy="modal-user">
-              {/* <strong className="has-text-success">Done</strong> */}
-              <strong className="has-text-danger">Planned</strong>
+              <strong
+                className={classNames(
+                  { 'has-text-success': completed },
+                  { 'has-text-danger': !completed },
+                )}
+              >
+                {completed
+                  ? (
+                    'Done'
+                  )
+                  : (
+                    'Planned'
+                  )}
```

Comment:
you can fit it in one line
```suggestion
                  ? 'Done'
                  : 'Planned'}
```